// @azure/cognitiveservices-newssearch/esm/operations/newsOperations.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/newsOperationsMappers";
import * as Parameters from "../models/parameters";
/** Class representing a NewsOperations. */
var NewsOperations = /** @class */ (function () {
    /**
     * Create a NewsOperations.
     * @param {NewsSearchClientContext} client Reference to the service client.
     */
    function NewsOperations(client) {
        this.client = client;
    }
    NewsOperations.prototype.search = function (query, options, callback) {
        return this.client.sendOperationRequest({
            query: query,
            options: options
        }, searchOperationSpec, callback);
    };
    NewsOperations.prototype.category = function (options, callback) {
        return this.client.sendOperationRequest({
            options: options
        }, categoryOperationSpec, callback);
    };
    NewsOperations.prototype.trending = function (options, callback) {
        return this.client.sendOperationRequest({
            options: options
        }, trendingOperationSpec, callback);
    };
    return NewsOperations;
}());
export { NewsOperations };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var searchOperationSpec = {
    httpMethod: "GET",
    path: "news/search",
    urlParameters: [
        Parameters.endpoint
    ],
    queryParameters: [
        Parameters.countryCode,
        Parameters.count,
        Parameters.freshness,
        Parameters.market,
        Parameters.offset,
        Parameters.originalImage,
        Parameters.query,
        Parameters.safeSearch,
        Parameters.setLang,
        Parameters.sortBy,
        Parameters.textDecorations,
        Parameters.textFormat
    ],
    headerParameters: [
        Parameters.xBingApisSDK,
        Parameters.acceptLanguage,
        Parameters.userAgent,
        Parameters.clientId,
        Parameters.clientIp,
        Parameters.location
    ],
    responses: {
        200: {
            bodyMapper: Mappers.News
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    serializer: serializer
};
var categoryOperationSpec = {
    httpMethod: "GET",
    path: "news",
    urlParameters: [
        Parameters.endpoint
    ],
    queryParameters: [
        Parameters.countryCode,
        Parameters.category,
        Parameters.count,
        Parameters.headlineCount,
        Parameters.market,
        Parameters.offset,
        Parameters.originalImage,
        Parameters.safeSearch,
        Parameters.setLang,
        Parameters.textDecorations,
        Parameters.textFormat
    ],
    headerParameters: [
        Parameters.xBingApisSDK,
        Parameters.acceptLanguage,
        Parameters.userAgent,
        Parameters.clientId,
        Parameters.clientIp,
        Parameters.location
    ],
    responses: {
        200: {
            bodyMapper: Mappers.News
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    serializer: serializer
};
var trendingOperationSpec = {
    httpMethod: "GET",
    path: "news/trendingtopics",
    urlParameters: [
        Parameters.endpoint
    ],
    queryParameters: [
        Parameters.countryCode,
        Parameters.count,
        Parameters.market,
        Parameters.offset,
        Parameters.safeSearch,
        Parameters.setLang,
        Parameters.since,
        Parameters.sortBy,
        Parameters.textDecorations,
        Parameters.textFormat
    ],
    headerParameters: [
        Parameters.xBingApisSDK,
        Parameters.acceptLanguage,
        Parameters.userAgent,
        Parameters.clientId,
        Parameters.clientIp,
        Parameters.location
    ],
    responses: {
        200: {
            bodyMapper: Mappers.TrendingTopics
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    serializer: serializer
};
