// @azure/cognitiveservices-newssearch/esm/models/mappers.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __assign } from "tslib";
export var ResponseBase = {
    serializedName: "ResponseBase",
    type: {
        name: "Composite",
        polymorphicDiscriminator: {
            serializedName: "_type",
            clientName: "_type"
        },
        uberParent: "ResponseBase",
        className: "ResponseBase",
        modelProperties: {
            _type: {
                required: true,
                serializedName: "_type",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var Identifiable = {
    serializedName: "Identifiable",
    type: {
        name: "Composite",
        polymorphicDiscriminator: ResponseBase.type.polymorphicDiscriminator,
        uberParent: "ResponseBase",
        className: "Identifiable",
        modelProperties: __assign(__assign({}, ResponseBase.type.modelProperties), { id: {
                readOnly: true,
                serializedName: "id",
                type: {
                    name: "String"
                }
            } })
    }
};
export var Response = {
    serializedName: "Response",
    type: {
        name: "Composite",
        polymorphicDiscriminator: ResponseBase.type.polymorphicDiscriminator,
        uberParent: "ResponseBase",
        className: "Response",
        modelProperties: __assign(__assign({}, Identifiable.type.modelProperties), { webSearchUrl: {
                readOnly: true,
                serializedName: "webSearchUrl",
                type: {
                    name: "String"
                }
            } })
    }
};
export var Thing = {
    serializedName: "Thing",
    type: {
        name: "Composite",
        polymorphicDiscriminator: ResponseBase.type.polymorphicDiscriminator,
        uberParent: "ResponseBase",
        className: "Thing",
        modelProperties: __assign(__assign({}, Response.type.modelProperties), { name: {
                readOnly: true,
                serializedName: "name",
                type: {
                    name: "String"
                }
            }, url: {
                readOnly: true,
                serializedName: "url",
                type: {
                    name: "String"
                }
            }, image: {
                readOnly: true,
                serializedName: "image",
                type: {
                    name: "Composite",
                    className: "ImageObject"
                }
            }, description: {
                readOnly: true,
                serializedName: "description",
                type: {
                    name: "String"
                }
            }, alternateName: {
                readOnly: true,
                serializedName: "alternateName",
                type: {
                    name: "String"
                }
            }, bingId: {
                readOnly: true,
                serializedName: "bingId",
                type: {
                    name: "String"
                }
            } })
    }
};
export var CreativeWork = {
    serializedName: "CreativeWork",
    type: {
        name: "Composite",
        polymorphicDiscriminator: ResponseBase.type.polymorphicDiscriminator,
        uberParent: "ResponseBase",
        className: "CreativeWork",
        modelProperties: __assign(__assign({}, Thing.type.modelProperties), { thumbnailUrl: {
                readOnly: true,
                serializedName: "thumbnailUrl",
                type: {
                    name: "String"
                }
            }, provider: {
                readOnly: true,
                serializedName: "provider",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Thing"
                        }
                    }
                }
            }, datePublished: {
                readOnly: true,
                serializedName: "datePublished",
                type: {
                    name: "String"
                }
            }, video: {
                readOnly: true,
                serializedName: "video",
                type: {
                    name: "Composite",
                    className: "VideoObject"
                }
            } })
    }
};
export var Article = {
    serializedName: "Article",
    type: {
        name: "Composite",
        polymorphicDiscriminator: ResponseBase.type.polymorphicDiscriminator,
        uberParent: "ResponseBase",
        className: "Article",
        modelProperties: __assign(__assign({}, CreativeWork.type.modelProperties), { wordCount: {
                readOnly: true,
                serializedName: "wordCount",
                type: {
                    name: "Number"
                }
            } })
    }
};
export var NewsArticle = {
    serializedName: "NewsArticle",
    type: {
        name: "Composite",
        polymorphicDiscriminator: ResponseBase.type.polymorphicDiscriminator,
        uberParent: "ResponseBase",
        className: "NewsArticle",
        modelProperties: __assign(__assign({}, Article.type.modelProperties), { category: {
                readOnly: true,
                serializedName: "category",
                type: {
                    name: "String"
                }
            }, headline: {
                readOnly: true,
                serializedName: "headline",
                type: {
                    name: "Boolean"
                }
            }, clusteredArticles: {
                readOnly: true,
                serializedName: "clusteredArticles",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "NewsArticle"
                        }
                    }
                }
            } })
    }
};
export var Answer = {
    serializedName: "Answer",
    type: {
        name: "Composite",
        polymorphicDiscriminator: ResponseBase.type.polymorphicDiscriminator,
        uberParent: "ResponseBase",
        className: "Answer",
        modelProperties: __assign(__assign({}, Response.type.modelProperties), { followUpQueries: {
                readOnly: true,
                serializedName: "followUpQueries",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Query"
                        }
                    }
                }
            } })
    }
};
export var SearchResultsAnswer = {
    serializedName: "SearchResultsAnswer",
    type: {
        name: "Composite",
        polymorphicDiscriminator: ResponseBase.type.polymorphicDiscriminator,
        uberParent: "ResponseBase",
        className: "SearchResultsAnswer",
        modelProperties: __assign(__assign({}, Answer.type.modelProperties), { totalEstimatedMatches: {
                readOnly: true,
                serializedName: "totalEstimatedMatches",
                type: {
                    name: "Number"
                }
            } })
    }
};
export var News = {
    serializedName: "News",
    type: {
        name: "Composite",
        polymorphicDiscriminator: ResponseBase.type.polymorphicDiscriminator,
        uberParent: "ResponseBase",
        className: "News",
        modelProperties: __assign(__assign({}, SearchResultsAnswer.type.modelProperties), { value: {
                required: true,
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "NewsArticle"
                        }
                    }
                }
            }, location: {
                readOnly: true,
                serializedName: "location",
                type: {
                    name: "String"
                }
            } })
    }
};
export var MediaObject = {
    serializedName: "MediaObject",
    type: {
        name: "Composite",
        polymorphicDiscriminator: ResponseBase.type.polymorphicDiscriminator,
        uberParent: "ResponseBase",
        className: "MediaObject",
        modelProperties: __assign(__assign({}, CreativeWork.type.modelProperties), { contentUrl: {
                readOnly: true,
                serializedName: "contentUrl",
                type: {
                    name: "String"
                }
            }, width: {
                readOnly: true,
                serializedName: "width",
                type: {
                    name: "Number"
                }
            }, height: {
                readOnly: true,
                serializedName: "height",
                type: {
                    name: "Number"
                }
            } })
    }
};
export var ImageObject = {
    serializedName: "ImageObject",
    type: {
        name: "Composite",
        polymorphicDiscriminator: ResponseBase.type.polymorphicDiscriminator,
        uberParent: "ResponseBase",
        className: "ImageObject",
        modelProperties: __assign(__assign({}, MediaObject.type.modelProperties), { thumbnail: {
                readOnly: true,
                serializedName: "thumbnail",
                type: {
                    name: "Composite",
                    className: "ImageObject"
                }
            } })
    }
};
export var Query = {
    serializedName: "Query",
    type: {
        name: "Composite",
        className: "Query",
        modelProperties: {
            text: {
                required: true,
                serializedName: "text",
                type: {
                    name: "String"
                }
            },
            displayText: {
                readOnly: true,
                serializedName: "displayText",
                type: {
                    name: "String"
                }
            },
            webSearchUrl: {
                readOnly: true,
                serializedName: "webSearchUrl",
                type: {
                    name: "String"
                }
            },
            searchLink: {
                readOnly: true,
                serializedName: "searchLink",
                type: {
                    name: "String"
                }
            },
            thumbnail: {
                readOnly: true,
                serializedName: "thumbnail",
                type: {
                    name: "Composite",
                    className: "ImageObject"
                }
            }
        }
    }
};
export var NewsTopic = {
    serializedName: "News/Topic",
    type: {
        name: "Composite",
        polymorphicDiscriminator: ResponseBase.type.polymorphicDiscriminator,
        uberParent: "ResponseBase",
        className: "NewsTopic",
        modelProperties: __assign(__assign({}, Thing.type.modelProperties), { isBreakingNews: {
                readOnly: true,
                serializedName: "isBreakingNews",
                type: {
                    name: "Boolean"
                }
            }, query: {
                readOnly: true,
                serializedName: "query",
                type: {
                    name: "Composite",
                    className: "Query"
                }
            }, newsSearchUrl: {
                readOnly: true,
                serializedName: "newsSearchUrl",
                type: {
                    name: "String"
                }
            } })
    }
};
export var TrendingTopics = {
    serializedName: "TrendingTopics",
    type: {
        name: "Composite",
        polymorphicDiscriminator: ResponseBase.type.polymorphicDiscriminator,
        uberParent: "ResponseBase",
        className: "TrendingTopics",
        modelProperties: __assign(__assign({}, Answer.type.modelProperties), { value: {
                required: true,
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "NewsTopic"
                        }
                    }
                }
            } })
    }
};
export var VideoObject = {
    serializedName: "VideoObject",
    type: {
        name: "Composite",
        polymorphicDiscriminator: ResponseBase.type.polymorphicDiscriminator,
        uberParent: "ResponseBase",
        className: "VideoObject",
        modelProperties: __assign(__assign({}, MediaObject.type.modelProperties), { motionThumbnailUrl: {
                readOnly: true,
                serializedName: "motionThumbnailUrl",
                type: {
                    name: "String"
                }
            }, motionThumbnailId: {
                readOnly: true,
                serializedName: "motionThumbnailId",
                type: {
                    name: "String"
                }
            }, embedHtml: {
                readOnly: true,
                serializedName: "embedHtml",
                type: {
                    name: "String"
                }
            }, allowHttpsEmbed: {
                readOnly: true,
                serializedName: "allowHttpsEmbed",
                type: {
                    name: "Boolean"
                }
            }, viewCount: {
                readOnly: true,
                serializedName: "viewCount",
                type: {
                    name: "Number"
                }
            }, thumbnail: {
                readOnly: true,
                serializedName: "thumbnail",
                type: {
                    name: "Composite",
                    className: "ImageObject"
                }
            }, videoId: {
                readOnly: true,
                serializedName: "videoId",
                type: {
                    name: "String"
                }
            }, allowMobileEmbed: {
                readOnly: true,
                serializedName: "allowMobileEmbed",
                type: {
                    name: "Boolean"
                }
            }, isSuperfresh: {
                readOnly: true,
                serializedName: "isSuperfresh",
                type: {
                    name: "Boolean"
                }
            } })
    }
};
export var Organization = {
    serializedName: "Organization",
    type: {
        name: "Composite",
        polymorphicDiscriminator: ResponseBase.type.polymorphicDiscriminator,
        uberParent: "ResponseBase",
        className: "Organization",
        modelProperties: __assign({}, Thing.type.modelProperties)
    }
};
export var ErrorModel = {
    serializedName: "Error",
    type: {
        name: "Composite",
        className: "ErrorModel",
        modelProperties: {
            code: {
                required: true,
                serializedName: "code",
                defaultValue: 'None',
                type: {
                    name: "String"
                }
            },
            subCode: {
                readOnly: true,
                serializedName: "subCode",
                type: {
                    name: "String"
                }
            },
            message: {
                required: true,
                serializedName: "message",
                type: {
                    name: "String"
                }
            },
            moreDetails: {
                readOnly: true,
                serializedName: "moreDetails",
                type: {
                    name: "String"
                }
            },
            parameter: {
                readOnly: true,
                serializedName: "parameter",
                type: {
                    name: "String"
                }
            },
            value: {
                readOnly: true,
                serializedName: "value",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var ErrorResponse = {
    serializedName: "ErrorResponse",
    type: {
        name: "Composite",
        polymorphicDiscriminator: ResponseBase.type.polymorphicDiscriminator,
        uberParent: "ResponseBase",
        className: "ErrorResponse",
        modelProperties: __assign(__assign({}, Response.type.modelProperties), { errors: {
                required: true,
                serializedName: "errors",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ErrorModel"
                        }
                    }
                }
            } })
    }
};
export var discriminators = {
    'ResponseBase.NewsArticle': NewsArticle,
    'ResponseBase.News': News,
    'ResponseBase.SearchResultsAnswer': SearchResultsAnswer,
    'ResponseBase.ImageObject': ImageObject,
    'ResponseBase.News/Topic': NewsTopic,
    'ResponseBase.Answer': Answer,
    'ResponseBase.Article': Article,
    'ResponseBase.Thing': Thing,
    'ResponseBase.Response': Response,
    'ResponseBase.TrendingTopics': TrendingTopics,
    'ResponseBase.VideoObject': VideoObject,
    'ResponseBase.CreativeWork': CreativeWork,
    'ResponseBase.Organization': Organization,
    'ResponseBase.Identifiable': Identifiable,
    'ResponseBase.ErrorResponse': ErrorResponse,
    'ResponseBase.MediaObject': MediaObject,
    'ResponseBase': ResponseBase
};
