// @azure/cognitiveservices-newssearch/esm/models/parameters.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
export var acceptLanguage = {
    parameterPath: [
        "options",
        "acceptLanguage"
    ],
    mapper: {
        serializedName: "Accept-Language",
        type: {
            name: "String"
        }
    }
};
export var category = {
    parameterPath: [
        "options",
        "category"
    ],
    mapper: {
        serializedName: "category",
        type: {
            name: "String"
        }
    }
};
export var clientId = {
    parameterPath: [
        "options",
        "clientId"
    ],
    mapper: {
        serializedName: "X-MSEdge-ClientID",
        type: {
            name: "String"
        }
    }
};
export var clientIp = {
    parameterPath: [
        "options",
        "clientIp"
    ],
    mapper: {
        serializedName: "X-MSEdge-ClientIP",
        type: {
            name: "String"
        }
    }
};
export var count = {
    parameterPath: [
        "options",
        "count"
    ],
    mapper: {
        serializedName: "count",
        type: {
            name: "Number"
        }
    }
};
export var countryCode = {
    parameterPath: [
        "options",
        "countryCode"
    ],
    mapper: {
        serializedName: "cc",
        type: {
            name: "String"
        }
    }
};
export var endpoint = {
    parameterPath: "endpoint",
    mapper: {
        required: true,
        serializedName: "Endpoint",
        defaultValue: 'https://api.cognitive.microsoft.com',
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export var freshness = {
    parameterPath: [
        "options",
        "freshness"
    ],
    mapper: {
        serializedName: "freshness",
        type: {
            name: "String"
        }
    }
};
export var headlineCount = {
    parameterPath: [
        "options",
        "headlineCount"
    ],
    mapper: {
        serializedName: "headlineCount",
        type: {
            name: "Number"
        }
    }
};
export var location = {
    parameterPath: [
        "options",
        "location"
    ],
    mapper: {
        serializedName: "X-Search-Location",
        type: {
            name: "String"
        }
    }
};
export var market = {
    parameterPath: [
        "options",
        "market"
    ],
    mapper: {
        serializedName: "mkt",
        type: {
            name: "String"
        }
    }
};
export var offset = {
    parameterPath: [
        "options",
        "offset"
    ],
    mapper: {
        serializedName: "offset",
        type: {
            name: "Number"
        }
    }
};
export var originalImage = {
    parameterPath: [
        "options",
        "originalImage"
    ],
    mapper: {
        serializedName: "originalImg",
        type: {
            name: "Boolean"
        }
    }
};
export var query = {
    parameterPath: "query",
    mapper: {
        required: true,
        serializedName: "q",
        type: {
            name: "String"
        }
    }
};
export var safeSearch = {
    parameterPath: [
        "options",
        "safeSearch"
    ],
    mapper: {
        serializedName: "safeSearch",
        type: {
            name: "String"
        }
    }
};
export var setLang = {
    parameterPath: [
        "options",
        "setLang"
    ],
    mapper: {
        serializedName: "setLang",
        type: {
            name: "String"
        }
    }
};
export var since = {
    parameterPath: [
        "options",
        "since"
    ],
    mapper: {
        serializedName: "since",
        type: {
            name: "Number"
        }
    }
};
export var sortBy = {
    parameterPath: [
        "options",
        "sortBy"
    ],
    mapper: {
        serializedName: "sortBy",
        type: {
            name: "String"
        }
    }
};
export var textDecorations = {
    parameterPath: [
        "options",
        "textDecorations"
    ],
    mapper: {
        serializedName: "textDecorations",
        type: {
            name: "Boolean"
        }
    }
};
export var textFormat = {
    parameterPath: [
        "options",
        "textFormat"
    ],
    mapper: {
        serializedName: "textFormat",
        type: {
            name: "String"
        }
    }
};
export var userAgent = {
    parameterPath: [
        "options",
        "userAgent"
    ],
    mapper: {
        serializedName: "User-Agent",
        type: {
            name: "String"
        }
    }
};
export var xBingApisSDK = {
    parameterPath: "xBingApisSDK",
    mapper: {
        required: true,
        isConstant: true,
        serializedName: "X-BingApis-SDK",
        defaultValue: 'true',
        type: {
            name: "String"
        }
    }
};
